src/plugins/integrations/PostgreIntegration.ts

```diff
@@ -41,66 +43,69 @@ class PostgreIntegration implements Integration {
-            [DBTags.DB_USER]: params.user,
-            [DBTags.DB_HOST]: params.host,
-            [DBTags.DB_PORT]: params.port,
-            [DBTags.DB_TYPE]: DBTypes.PG,
-            [SpanTags.TOPOLOGY_VERTEX]: true,
-            [SpanTags.TRIGGER_DOMAIN_NAME]: LAMBDA_APPLICATION_DOMAIN_NAME,
-            [SpanTags.TRIGGER_CLASS_NAME]: LAMBDA_APPLICATION_CLASS_NAME,
-            [SpanTags.TRIGGER_OPERATION_NAMES]: [tracer.functionName],
+        try {
+          const tracer = ThundraTracer.getInstance();
+          const parentSpan = tracer.getActiveSpan();
+
+          const params = this.connectionParameters;
+          const span = tracer._startSpan(params.database, {
+            childOf: parentSpan,
+            domainName: DomainNames.DB,
+            className: ClassNames.RDB,
+            disableActiveStart: true,
           });
-        }
 
-        const pgQuery = query.apply(this, arguments);
+          if (params) {
+            span.addTags({
+              [SpanTags.SPAN_TYPE]: SpanTypes.RDB,
+              [DBTags.DB_INSTANCE]: params.database,
+              [DBTags.DB_USER]: params.user,
+              [DBTags.DB_HOST]: params.host,
+              [DBTags.DB_PORT]: params.port,
+              [DBTags.DB_TYPE]: DBTypes.PG,
+              [SpanTags.TOPOLOGY_VERTEX]: true,
+              [SpanTags.TRIGGER_DOMAIN_NAME]: LAMBDA_APPLICATION_DOMAIN_NAME,
+              [SpanTags.TRIGGER_CLASS_NAME]: LAMBDA_APPLICATION_CLASS_NAME,
+              [SpanTags.TRIGGER_OPERATION_NAMES]: [tracer.functionName],
+            });
+          }
 
-        let statement = pgQuery.text;
-        statement = Utils.replaceArgs(statement, pgQuery.values);
+          const pgQuery = query.apply(this, arguments);
 
-        if (statement) {
-          const statementType = statement.split(' ')[0].toUpperCase();
-          span.addTags({
-            [DBTags.DB_STATEMENT_TYPE]: statementType,
-            [DBTags.DB_STATEMENT]: statement,
-            [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ? SQLQueryOperationTypes[statementType] : 'READ',
-          });
-        }
+          let statement = pgQuery.text;
+          statement = Utils.replaceArgs(statement, pgQuery.values);
 
-        const originalCallback = pgQuery.callback;
-
-        pgQuery.callback = (err: any, res: any) => {
-          if (err) {
-            const parseError = Utils.parseError(err);
-            span.setTag('error', true);
-            span.setTag('error.kind', parseError.errorType);
-            span.setTag('error.message', parseError.errorMessage);
+          if (statement) {
+            const statementType = statement.split(' ')[0].toUpperCase();
+            span.addTags({
+              [DBTags.DB_STATEMENT_TYPE]: statementType,
+              [DBTags.DB_STATEMENT]: statement,
+              [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ? SQLQueryOperationTypes[statementType] : 'READ',
```

Comment:
we might return empty operation type instead of `READ` if the operation is not known

Reply:
Done.